File: src/leetcode/python/MaximumSubarray_53.py
class MaximumSubarray_53(object):
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        sub_max = mx = nums[0]
        for i in range(1, len(nums)):
            mx = max(max(sub_max+nums[i], nums[i]), max)
        return mx

File: src/leetcode/java/MaximumSubarray_53.java
package leetcode.java;

public class MaximumSubarray_53 {
  public int maxSubArray(int[] nums) {
    int max = nums[0], subMax = nums[0];
    for (int i = 1; i < nums.length; i++) {
      max = Math.max(Math.max(subMax + nums[i], nums[i]), max);
    }
    return max;
  }
}
